### PR TITLE
Core/ObjectMgr: Fix StatCount exceeding existing ItemStat fields.

### LINKED ISSUE #15868: [3.3.5] Wrong Item StatsCount generate crash in ObjectMgr.cpp
StatsCount is limited to MAX_ITEM_PROTO_STATS=10 (in 3.3.5)
A much higher value can generate a crash in ObjectMgr.cpp here (at line 2362):
```
        for (uint8 i = 0; i < itemTemplate.StatsCount; ++i)
        {
            itemTemplate.ItemStat[i].ItemStatType  = uint32(fields[28 + i*2].GetUInt8());
            itemTemplate.ItemStat[i].ItemStatValue = int32(fields[29 + i*2].GetInt16());
        }
```
The loop is creating a buffer overflow if StatsCount is > 10

The reason is because the check is done AFTER that loop some codelines later (line 2600):
```
        if (itemTemplate.StatsCount > MAX_ITEM_PROTO_STATS)
        {
            TC_LOG_ERROR("sql.sql", "Item (Entry: %u) has too large value in statscount (%u), replace by hardcoded limit (%u).", entry, itemTemplate.StatsCount, MAX_ITEM_PROTO_STATS);
            itemTemplate.StatsCount = MAX_ITEM_PROTO_STATS;
        }
```

Solution: Correct the value of StatsCount before the first loop.

### PR DESCRIPTION
**Pull request by EgoSumThan**
Closes #15868

### CODE REVIEW

**Comment by tkrokli** [comment]
Suggestion for a commit text (2 lines below the title in git) and in your description post here:
```
In the 3.3.5 game client, StatsCount is limited to MAX_ITEM_PROTO_STATS=10
and the value check is done some codelines later, AFTER the 'for' loop.
A much higher value can generate a crash in ObjectMgr.cpp
because it will create a buffer overflow if StatsCount is > 10.

The solution is to correct the value of StatsCount in the 'for' loop.

Closes #15868
```

**Comment by MitchesD** [comment]
@tkrokli I appreciate that you want make things more clear, but sometimes it really looks you want to make people look stupid

**Comment by Ascathor** [comment]
@tkrokli Your explaination is good and informative, but I do not really see the need to a lengthy breakdown of literally 1 line change and 2 lines added for comments :p No hate friend

File: src/server/game/Globals/ObjectMgr.cpp [hunk 1]
```diff
@@ -2347,7 +2347,8 @@ void ObjectMgr::LoadItemTemplates()
         itemTemplate.ContainerSlots            = uint32(fields[26].GetUInt8());
         itemTemplate.StatsCount                = uint32(fields[27].GetUInt8());
 
-        for (uint8 i = 0; i < itemTemplate.StatsCount; ++i)
+        // Loops according to StatsCount, but never exceeds MAX_ITEM_PROTO_STATS (Client-side limited to 10)
+        for (uint8 i = 0; i < itemTemplate.StatsCount && i <= MAX_ITEM_PROTO_STATS; ++i)
```
Comment (jackpoz) [inline src/server/game/Globals/ObjectMgr.cpp]: it's makes little sense to check i <= MAX_ITEM_PROTO_STATS and check it again later

**Pull request closed without merging**